### DISCUSSION TITLE
Tomcat Not Working

### DISCUSSION THREAD

**Discussion by ccentrella** [apache/netbeans#4535] [Q&A]
Hello,

I cannot get Tomcat 9 (or 8.5 or 10) working with NetBeans. I installed the Tomcat and when I try to start the server a dialog pops up asking for my username and password. If I click okay, the dialog comes back and after a few times I get this error in the console:

`25-Aug-2022 10:25:41.899 WARNING [http-nio-8080-exec-6] org.apache.catalina.realm.LockOutRealm.filterLockedAccounts An attempt was made to authenticate the locked user [admin]`

I tried clicking cancel and that doesn't work. I'm certain that the username and password is correct, so I'm not sure what is happening, and my password consists of only letters and numbers so it isn't caused by special characters in my password. I read in one place that the Tomcat installation cannot run simultaneously as a service on Windows and in Netbeans, so I opened Tomcat9w.exe, stopped the service, and restarted Netbeans. No luck.  I tried running Netbeans as Administrator, but that doesn't make a difference either. When this happen, Tomcat seems to start anyway, in some weird state. When I try to stop it, the same thing happens, and I haven't been able to stop it.

At this point, there seems to be a bug in Netbeans. This makes no sense and most of the discussions I've found say that this issue has been fixed in a previous version of NetBeans, but I'm using the latest version of Netbeans 14.

Here are the contents of tomcat-users.xml:

```
<!-- 
  Licensed to the Apache Software Foundation (ASF) under one or more
  contributor license agreements.  See the NOTICE file distributed with
  this work for additional information regarding copyright ownership.
  The ASF licenses this file to You under the Apache License, Version 2.0
  (the "License"); you may not use this file except in compliance with
  the License.  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 -->
<tomcat-users xmlns="http://tomcat.apache.org/xml" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="1.0" xsi:schemaLocation="http://tomcat.apache.org/xml tomcat-users.xsd">
<!-- 
  By default, no user is included in the "manager-gui" role required
  to operate the "/manager/html" web application.  If you wish to use this app,
  you must define such a user - the username and password are arbitrary.

  Built-in Tomcat manager roles:
    - manager-gui    - allows access to the HTML GUI and the status pages
    - manager-script - allows access to the HTTP API and the status pages
    - manager-jmx    - allows access to the JMX proxy and the status pages
    - manager-status - allows access to the status pages only

  The users below are wrapped in a comment and are therefore ignored. If you
  wish to configure one or more of these users for use with the manager web
  application, do not forget to remove the <!.. ..> that surrounds them. You
  will also need to set the passwords to something appropriate.
 -->
<!-- 
  <user username="admin" password="<must-be-changed>" roles="manager-gui"/>
  <user username="robot" password="<must-be-changed>" roles="manager-script"/>
 -->
<!-- 
  The sample user and role entries below are intended for use with the
  examples web application. They are wrapped in a comment and thus are ignored
  when reading this file. If you wish to configure these users for use with the
  examples web application, do not forget to remove the <!.. ..> that surrounds
  them. You will also need to set the passwords to something appropriate.
 -->
<!-- 
  <role rolename="tomcat"/>
  <role rolename="role1"/>
  <user username="tomcat" password="<must-be-changed>" roles="tomcat"/>
  <user username="both" password="<must-be-changed>" roles="tomcat,role1"/>
  <user username="role1" password="<must-be-changed>" roles="role1"/>
 -->
<user password="[my-password]" roles="manager-gui,admin-gui, manager-script,admin" username="admin"/>
</tomcat-users>
```
Has anyone experienced a similar problem?

**Comment by 6PATyCb** [comment]
Hi, I am using tomcat with netbeans every day. When required install new version of Tomcat, i just download Tomcat as archive, extract it as `c:\Users\6PATyCb\Documents\tomcats\apache-tomcat-9.0.58\` and then start wizard to add Tomcat in NetBeans:
![image](https://user-images.githubusercontent.com/33724393/194770514-f340ca90-e7ca-4968-84fe-cc3862f61613.png)

I don't know how exactly is worked in Netbeans, but after installation you don't need to change tomcat-users.xml. Only input tomcat:tomcat on popup window at first start.

**Comment by huynd4104** [comment]
sorry i am having this problem too did you find a solution, 2 years have passed but hope you still remember the solution
